etc/home-env-funcs.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+sirepo_add_admin() {
+    declare u
+    # POSIT: only uids in user and they don't have any specials
+    for u in $(ls ~/src/radiasoft/sirepo/run/user); do
+        sirepo roles add "$u" adm
+    done
+}
+
+sirepo_mail() {
+    # xargs trims whitespace
+    grep 'http://' $(ls -dt ~/mail/* | head -n1) | xargs
```

Comment:
I was curious if perl would be better, and I'm not sure it is:

```perl
perl <<'EOF'
use strict;
chdir($ENV{SIREPO_DEV_MAIL_DIR}) || die;
my($x) = [sort({$b <=> $a} grep(!/\D/, glob('*')))];
if (@$x) {
    open(IN, $x->[0]) || die;
    print((map(m{(http://\S+auth\S+)} ? $1 : (), <IN>))[0], "\n");
}
EOF
```

The shell pipeline definitely keeps things simple. I don't like the xargs comment though. xargs implicitly uses echo as the program. Maybe something like:

```sh
perl -n -e 'm{(http://\S+)} && print("$1\n")' $(ls -dt ~/mail/* | head -n1)
```

You could do the same thing with `sed`.

Reply:
Definitely not better to my untrained perl eyes:)

The second option is fine.

9cf9e4cee